mcu/src/mcu/uart_reader.py

```diff
@@ -0,0 +1,455 @@
+"""
+uart-reader.py
+ELEC PROJECT - 210x
+"""
+
+import os
+import signal
+import sys
+import threading
+import tkinter as tk
+from tkinter import messagebox, scrolledtext, ttk
+
+import click
+import numpy as np
+import plotly.graph_objects as go
+import plotly.io as pio
+import serial
+import soundfile as sf
+from plotly.subplots import make_subplots
+from serial.tools import list_ports
+
+###################### MODIFY BELOW THIS LINE #####################
+####################################################################
+
+# Default values
+PRINT_PREFIX = "SND:HEX:"
+FREQ_SAMPLING = 10200
+DEFAULT_BAUDRATE = 115200
+VAL_MAX_ADC = 4096
+VDD = 3.3
+LOG_ENABLED = False
+AUTO_DELETE_AUDIO = False
+DEFAULT_PORT_INDEX = 1  # Change this if the default port is not the first one
+
+###################### DO NOT MODIFY BELOW THIS LINE #####################
+##########################################################################
+
+
+def generate_audio(buf, file_name):
+    buf = np.asarray(buf, dtype=np.float64)
+    buf = buf - np.mean(buf)
+    buf /= max(abs(buf))
+    sf.write("audio_files/" + file_name + ".wav", buf, FREQ_SAMPLING)
```

Comment:
Given its function, it looks that 'write_audio' may be more appropriate for a name.